packages/eslint-plugin/utils.js

```diff
@@ -60,6 +60,51 @@ function isDependencyBlock( jsdoc ) {
 	);
 }
 
+function isImported( node ) {
+	if ( ! node || ! node.type ) {
+		return false;
+	}
+
+	const importTypes = [
+		'ImportDefaultSpecifier',
+		'ImportSpecifier',
+		'ExportDefaultDeclaration',
+		'ExportDeclaration',
+		'ImportDeclaration',
+	];
+
+	if ( importTypes.includes( node.type ) || importTypes.includes( node?.parent?.type ) ) {
+		return true;
+	}
+
+	for ( const attribute in [ 'parent', 'init', 'imported' ] ) {
+		if ( node[ attribute ] ) {
+			return isImported( node[ attribute ] );
+		}
+	}
+
+	if (
+		node.declarations &&
+		node.declarations.length
+	) {
```

Comment:
Let's simplify this here as well
```suggestion
	if ( node.declarations?.length ) {
```